v6_agent_guides/Payment_Component_Public_API_Migration_Guide.md

```diff
@@ -0,0 +1,501 @@
+# Integrating Payment Components with AdyenCheckout
+
+This guide explains how to integrate a payment component with the new `AdyenCheckout` module, enabling it to work with `CheckoutConfiguration` and receive proper theme propagation.
+
+## Overview
+
+The `AdyenCheckout` module provides a unified way to configure and create payment components through `CheckoutConfiguration`. This architecture enables:
+
+- Centralized theme management via `CheckoutConfiguration.theme()`
+- Consistent component creation through `CheckoutComponentBuilder`
+- Type-safe configuration storage and retrieval
+
+```
+CheckoutConfiguration (public API)
+    │
+    ├── theme: AdyenTheme
+    ├── configurations: [ComponentType: Configuration]
+    │
+    └── CheckoutComponentBuilder.build(for: paymentMethod)
+            │
+            ├── Retrieves or creates default configuration
+            ├── Propagates theme: config.theme = checkoutConfig.theme
+            │
+            └── ComponentFactory.create() → PaymentComponent
+```
+
+---
+
+## Part 1: Main Codebase Changes
+
+### Step 1: Conform Configuration to `CheckoutComponentConfiguration`
+
+Update your component's `Configuration` struct to conform to the protocol:
+
+```swift
+// Before
+public struct Configuration: AnyXXXConfiguration, AnyPersonalInformationConfiguration {
+    public var style: FormComponentStyle
+    internal let showsSubmitButton: Bool
+    // ...
+}
+
+// After
+public struct Configuration: AnyXXXConfiguration,
+                            AnyPersonalInformationConfiguration,
+                            CheckoutComponentConfiguration {  // Add protocol
+
+    package let componentType: CheckoutComponentType = .payment(.yourPaymentType)
+    package var theme: AdyenTheme = .init()
+    package var showsSubmitButton: Bool  // Change: internal let → package var
+
+    public var style: FormComponentStyle
+    // ...
+}
+```
+
+**Key changes:**
+- Add `CheckoutComponentConfiguration` conformance
```

Comment:
We may need to mention the config may be nested and need to be moved out, while also turning the component package.